pkg/block/azure/main_test.go

```diff
@@ -0,0 +1,118 @@
+package azure_test
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob"
+	"github.com/benburkert/dns"
+	"github.com/ory/dockertest/v3"
+)
+
+const (
+	azuriteContainerTimeoutSeconds = 10 * 60 // 10 min
+	containerName                  = "container1"
+	accountName                    = "account1"
+	accountKey                     = "key1"
+)
+
+var (
+	pool     *dockertest.Pool
+	blockURL string
+)
+
+func createDNSServer(ctx context.Context, domain string) {
+	zone := &dns.Zone{
+		Origin: domain + ".",
+		TTL:    5 * time.Minute,
+		RRs: dns.RRSet{
+			accountName: map[dns.Type][]dns.Record{
+				dns.TypeA: {
+					&dns.A{A: net.IPv4(127, 0, 0, 1).To4()},
+				},
+			},
+		},
+	}
+
+	srv := &dns.Server{
+		Addr:    ":53",
```

Comment:
I think binds to a specific port on the host machine.  But that port may be occupied, maybe someone runs dnsmasq!  Also because this uses a port <1024 it has to run as root on some systems.

Consider binding only on "127.0.0.1:0" or similar, and then _querying_ the server which ephemeral port it received.  Not sure you can do it with this DNS server package though.  I _think_ that [this dns.Server](https://pkg.go.dev/github.com/miekg/dns#Server) might be able to bind an ephemeral port if you configure it with `Addr: "127.0.0.1:0"`, and then fetch that port using `net.SplitHostPort(server.Listener.Addr().String())`.